packages/gatsby-source-contentful/src/gatsby-node.js

```diff
@@ -121,24 +119,53 @@ exports.sourceNodes = async (
   // are "updated" so will get the now deleted reference removed.
 
   function deleteContentfulNode(node) {
-    const localizedNodes = locales
-      .map(locale => {
-        const nodeId = createNodeId(
-          normalize.makeId({
-            spaceId: space.sys.id,
-            id: node.sys.id,
-            currentLocale: locale.code,
-            defaultLocale,
+    locales.forEach(locale => {
+      const nodeId = createNodeId(
+        normalize.makeId({
+          spaceId: space.sys.id,
+          id: node.sys.id,
+          currentLocale: locale.code,
+          defaultLocale,
+        })
+      )
+      const gatsbyNode = getNode(nodeId)
+
+      if (gatsbyNode) {
```

Comment:
Maybe return early? Because I think this `if` block doesn't contain the `touchNode()` and `deleteNode()` calls below? Before they would be conditional because of the `.filter()` so now that's no longer the case. Maybe I'm wrong and they are in the block, but a return early here would probably be best, anyways?